sphinx/source/docs/user_guide/examples/styling_specifying_colors.py

```diff
@@ -1,52 +1,39 @@
-from itertools import product
-from math import pi
-
-from bokeh.io import output_file, show
-from bokeh.plotting import figure
-
-output_file('properties_alpha.html')
-
-cats = ['None', 'Alpha', 'RGB', 'RGBA', 'Alpha+RGB', 'Alpha+RGBA']
-
-p = figure(x_range=cats, y_range=cats,
-           title="Fill and Line Color Combinations")
-
-p.xaxis.axis_label = "Fill Options"
-p.xaxis.major_label_orientation = pi/4
-p.yaxis.axis_label = "Line Options"
-p.grid.grid_line_color = None
-p.toolbar_location = None
-
-alpha = 0.5
-fill_color = (242, 44, 64)
-fill_color_alpha = (242, 44, 64, alpha)
-line_color = (0, 0, 0)
-line_color_alpha = (0, 0, 0, alpha)
-
-# define fill and line color combinations
-fill = [(0, {}),
-        (1, {'fill_alpha': alpha}),
-        (2, {'fill_color': fill_color}),
-        (3, {'fill_color': fill_color_alpha}),
-        (4, {'fill_alpha': alpha, 'fill_color': fill_color}),
-        (5, {'fill_alpha': alpha, 'fill_color': fill_color_alpha})]
-
-line = [(0, {}),
-        (1, {'line_alpha': alpha}),
-        (2, {'line_color': line_color}),
-        (3, {'line_color': line_color_alpha}),
-        (4, {'line_alpha': alpha, 'line_color': line_color}),
-        (5, {'line_alpha': alpha, 'line_color': line_color_alpha})]
-
-# plot intersection of fill and line combinations
-combinations = product(fill, line)
-for comb in combinations:
-    x, fill_options = comb[0]
-    y, line_options = comb[1]
-
-    options = fill_options.copy()
-    options.update(line_options)
-
-    p.circle([cats[x]], [cats[y]], line_width=7, size=50, **options)
+import numpy as np
+
+from bokeh.plotting import figure, show, output_file
+
+output_file("specifying_colors.html")
+
+x = [1, 2, 3]
+y1 = [1, 4, 2]
+y2 = [2, 1, 4]
+y3 = [4, 3, 2]
+
+# use a single RGBA color
+single_color = (255, 0, 0, 0.5)
+
+# use a list of HSL colors
+list_of_colors = [
+    "hsl(60deg 100% 50% / 1.0)",
+    "hsl(0deg 100% 50% / 0.4)",
+    "hsl(120deg 100% 50% / 1.0)",
```

Comment:
instead of 3 hsl values, what about a named color value, an "rgb(...)" value, and a hsl value?

Reply:
Good idea!